### DISCUSSION TITLE
Asset catalogs are not available for the iOS platform. - Build error for iOS but not Android

### DISCUSSION THREAD

**Discussion by DclearEE** [unoplatform/uno#10348] [Q&A] [answered]
Hi Uno Community. I am getting an error specific to iOS and not anything else. I can build and deploy the project for Android.

During the build after the signing of the app with my paired remote Mac, I get an error:

```

1>Tool arch execution finished (exit code = 1).
1><?xml version="1.0" encoding="UTF-8"?>
1><!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
1><plist version="1.0">
1><dict>
1>	<key>com.apple.actool.errors</key>
1>	<array>
1>		<dict>
1>			<key>description</key>
1>			<string>Asset catalogs are not available for the iOS platform.</string>
1>		</dict>
1>	</array>
1></dict>
1></plist>
1>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.4.446-ci.-release-6-0-4xx.446\tools\msbuild\iOS\Xamarin.Shared.targets(842,3): error : arch exited with code 1
1>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.4.446-ci.-release-6-0-4xx.446\tools\msbuild\iOS\Xamarin.Shared.targets(842,3): error :
1>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.4.446-ci.-release-6-0-4xx.446\tools\msbuild\iOS\Xamarin.Shared.targets(842,3): error : actool exited with code 1
1>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.4.446-ci.-release-6-0-4xx.446\tools\msbuild\iOS\Xamarin.Shared.targets(842,3): error :
1>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.4.446-ci.-release-6-0-4xx.446\tools\msbuild\iOS\Xamarin.Shared.targets(842,3): error : Asset catalogs are not available for the iOS platform.
1>Done building project "SkottieSample.Mobile.csproj" -- FAILED.
```

I have assets in the "Asset" folder, organized into their own folders. All assets are set to "Build Action: Content. Copy to Output: Do not copy".
Is there something I'm missing as far as storing content for an iOS build?
For reference this is the same project from my last question I posed here: [https://github.com/unoplatform/uno/discussions/10346](https://github.com/unoplatform/uno/discussions/10346)

Please help me understand this error and how to fix it. 
Thank you!

**Comment by jeromelaban** [comment]
Thanks for the report. Can you provide a repro project for this issue?

**Reply by Xiaoy312** [reply]
I manage to get this error today, accidently by changing Xcode path in the settings:
```
VS (windows)\Tools\Options > Xamarin\iOS Settings\Apple SDK\Xcode path: `/Applications/Xcode-14.2.app` -> `/Applications/Xcode.app`
```
This error happens if you try to deploy/debug the ios app, but only if you dont restart visual studio after changing that setting.
The project in question is a mostly fresh `unoapp` (Uno.ProjectTemplates.Dotnet@4.7.44) with only cs+xaml changes.

**Comment by DclearEE** [comment, marked as answer]
The problem was fixed when I switched some YAML Deserialization functions to NewtonSoft.JSON functions. And converted the YAML files to JSON files. Unfortunately I was not able to debug this problem on iOS but I was able to find the bug by debugging WASM. I think it's just a coincidence that it was the same bug.

I was using the package YamlDotNet version 12.0.2 in this project. While it was building for all platforms, only Windows and Android were able to successfully deploy and run. 

That's what I get for trying to be fancy and use YAML lol. I think JSON is converted faster than YAML anyway.
